mcbackend/adapters/hopsy.py

```diff
@@ -0,0 +1,76 @@
+import numpy
+import numpy as np
+
+import hopsy
+from typing import Dict, List, Optional, Union
+import hagelkorn
+
+from mcbackend.meta import Coordinate, DataVariable, Variable
+
+from ..core import Backend, Chain, Run, RunMeta
+
+
+class TraceBackend(hopsy.Backend):
+    """Adapter to create a hospy backend from any McBackend."""
+
+    supports_sampler_stats = True
+
+    def __init__(self, backend: Backend, name: str = None):
+        super().__init__(name)
+        self.run_id = hagelkorn.random(digits=6)
+        print(f"Backend run id: {self.run_id}")
+        self._backend: Backend = backend
+
+        # Names
+        self.var_names = None
```

Comment:
You don't really need this, because your iteration in `record` can get the same information from `self._run.meta.variables`